packages/api/test/functional/announcement.spec.js

```diff
@@ -0,0 +1,347 @@
+	await pendingdAnnouncement.user().associate(user);
+	await pendingdAnnouncement.institution().associate(institution);
+
+	const response = await client.get(`/announcements/${publishedAnnouncement.id}`).end();
+
+	response.assertStatus(200);
+	response.assertJSONSubset(publishedAnnouncement.toJSON());
+
+	const response2 = await client.get(`/announcements/${pendingdAnnouncement.id}`).end();
+	response2.assertStatus(400);
+	response2.assertJSONSubset(
+		errorPayload(
+			errors.RESOURCE_NOT_FOUND,
+			antl('error.resource.resourceNotFound', { resource: 'Announcement' }),
+		),
+	);
+});
+
+test('GET /announcements/:id returns announcement if logged user is owner, otherwise only published', async ({
+	client,
+}) => {
+	const { user: ownerUser } = await createUser({ append: { status: 'verified' } });
+	const { user: otherUser } = await createUser({ append: { status: 'verified' } });
+	const institution = await Factory.model('App/Models/Institution').create();
+
+	const pendingdAnnouncement = await Factory.model('App/Models/Announcement').create();
+	await pendingdAnnouncement.user().associate(ownerUser);
+	await pendingdAnnouncement.institution().associate(institution);
+
+	const response = await client
+		.get(`/announcements/${pendingdAnnouncement.id}`)
+		.loginVia(ownerUser, 'jwt')
+		.end();
+
+	response.assertStatus(200);
+	response.assertJSONSubset(pendingdAnnouncement.toJSON());
+
+	const response2 = await client
+		.get(`/announcements/${pendingdAnnouncement.id}`)
+		.loginVia(otherUser, 'jwt')
+		.end();
+	response2.assertStatus(400);
+	response2.assertJSONSubset(
+		errorPayload(
+			errors.RESOURCE_NOT_FOUND,
+			antl('error.resource.resourceNotFound', { resource: 'Announcement' }),
+		),
+	);
+});
+
+test('POST /announcements creates a new Announcement', async ({ client, assert }) => {
+	const { user } = await createUser({ append: { status: 'verified' } });
+	const keywordTaxonomy = await Taxonomy.getTaxonomy('KEYWORDS');
+	const keywordTerms = await Factory.model('App/Models/Term').createMany(5);
+	await keywordTaxonomy.terms().saveMany(keywordTerms);
+	const keywordTermsIds = keywordTerms.map((keyword) => keyword.id);
+
+	const targetAudienceTerms = await Taxonomy.getTaxonomyTerms('TARGET_AUDIENCE');
+	const targetAudienceTermsIds = targetAudienceTerms.rows.map(
+		(targetAudience) => targetAudience.id,
+	);
+
+	const institution = await Factory.model('App/Models/Institution').create();
+
+	const response = await client
+		.post('/announcements')
+		.loginVia(user, 'jwt')
+		.send({
+			institution_id: institution.id,
+			announcement_number: '001/2021',
+			title: 'First 2021 Announcement',
+			description: 'First 2021 Announcement description',
+			keywords: keywordTermsIds,
+			targetAudiences: targetAudienceTermsIds,
+			financial_resources: 1000,
+			comment: 'single comment',
+			start_date: '2021-01-01',
+			end_date: '2021-01-31',
+			url: 'http://first2021annoucement.com',
+		})
```

Comment:
Vc pode substituir isso por um `const announcement = await Factory.model('App/Models/Announcement').make();`
```suggestion
		.send(announcement.toJSON())
```